packages/capabilities/readme.md

```diff
@@ -25,3 +33,126 @@ import * as Voucher from '@web3-storage/capabilities/voucher'
 
 // This package has a "main" entrypoint but we recommend the usage of the specific imports above
 ```
+
+### Capability types
+
+The capability objects exported by this package are defined using ucanto's type-inference based capability parser. This results in concrete types that capture the details of each capability, allowing type-safe invocation and validation. 
+
+When inspecting the concrete types of a capability object (e.g. in your IDE), you may see something similar to the following:
+
+
+```ts
+const add: TheCapabilityParser<DerivedMatch<{
+    can: "store/add";
+    with: URI<"did:">;
+    nb: InferCaveats<{
+        link: typeof Store.Schema.Link;
+        size: Store.Schema.NumberSchema<number & Phantom<{
+            typeof: "integer";
+        }>, unknown>;
+        origin: Store.Schema.Schema<...>;
+    }>;
+}, CapabilityMatch<...> | DerivedMatch<...>>>
+```
+
+While this is a fairly complex type signature, most of the types exist to support the mechanics of the capability parser and can generally be ignored when using the capabilities. The most interesting part as a user is the definition in the `DerivedMatch` type constraint, which shows the inferred ability and the constraints upon the resource URI and the caveats. In the example above, the `can` field shows that this capability's ability is `"store/add"`, its resource URI (the `with` field) must have the `"did:"` scheme, and there are three caveats defined in the `nb` field: `link`, `size`, and `origin`, each of which have constraints on their allowed values.
+
+### Using the exported capabilities
+
+The capability object exposes three methods via the `TheCapabilityParser` interface: `create`, `invoke`, and `delegate`. 
```

Comment:
this could be good docs to have in ucanto itself (too) https://github.com/web3-storage/ucanto/